c2cgeoform/static/src/controls.js

```diff
@@ -39,3 +41,36 @@ export function addControls(options) {
   })
   target.appendChild(container)
 }
+
+export function addGeolocation(view) {
+  // getCurrentPosition is only available with HTTPS
+  if (location.protocol !== 'https:' && location.hostname !== 'localhost') {
+    console.warn('Geolocation is only available with HTTPS protocol')
+    return
+  }
+
+  let button = $('<div/>', {
+    class: 'ol-control c2cgeoform-locate-me-btn',
+  }).append(
+    $('<button/>', { type: 'button' }).append(
+      $('<i/>', { class: 'glyphicon glyphicon-record' })
+    )
+  )
+  $('.map').append(button)
```

Comment:
if map was the function param you could access view & target element without having to query whole document with selector. Would be more change-proof.